src/utils/validateUin.tsx

```diff
@@ -0,0 +1,43 @@
+import { ERROR_MESSAGE } from "../context/alert";
+
+const validate = (cleanedInputUin: string, idRegex: string): boolean => {
+  // returns false if it fails regex
+  if (!cleanedInputUin.match(idRegex)) {
+    return false;
+  }
+  // if regex passes, prefix can be removed for checksum calculation
+  const uinWithoutPrefix = cleanedInputUin.slice(4);
+
+  const checksum =
+    uinWithoutPrefix
+      .split("")
+      .slice(0, -1)
+      .map(Number)
+      .reduce((prev, curr) => prev + curr) % 8;
+
+  if (
+    checksum !=
+    uinWithoutPrefix.charCodeAt(uinWithoutPrefix.length - 1) - 65
+  ) {
+    return false;
+  }
+
+  return true;
+};
+
+/**
+ * This function is called to clean the input UIN and
+ * validate the specific UIN using the regex check
+ * and our checksum.
```

Comment:
can be more generic here 😂 
```suggestion
 * and checksum.
```

Reply:
changed!